extensions/cornerstone-3d/src/services/ViewportService/Cornerstone3DViewportService.ts

```diff
@@ -249,17 +283,147 @@ class Cornerstone3DViewportService implements IViewportService {
+    // If you call loadStudyMetadata and it's not in the DicomMetadataStore cache, it should fire
+    // a request through the data source?
+    // (This call may or may not create sub-requests for series metadata)
+    //
+    // If you call loadSeriesMetadata and a request is ongoing for the study, you have to wait for
+    // that? Or should it force a separate request? May be slower?
+    // For the moment I just stuck this whole operation in a setTimeout so we are sure the
+    // metadata has arrived
+    const viewportVolumeInputs = {};
+    const displaySetOptionsArray = viewportInfo.getDisplaySetOptions();
+
+    for (let i = 0; i < viewportData.imageIds.length; i++) {
+      const imageIds = viewportData.imageIds[i];
+      const displaySetInstanceUID = viewportData.displaySetInstanceUIDs[i];
+      const displaySetOptions = displaySetOptionsArray[i];
+
+      // We need to get the volumeId again here since, for toolGroups
+      // right now we only have one volume per viewport, but for
+      // rendering we can have multiple volumes per viewport
+      const volumeId = displaySetInstanceUID;
+
+      // if (displaySet.needsRerendering) {
+      //   console.warn('Removing volume from cache', volumeId);
+      //   cache.removeVolumeLoadObject(volumeId);
+      //   displaySet.needsRerendering = false;
+      //   this.displaySetsNeedRerendering.add(displaySet.displaySetInstanceUID);
+      // }
+
+      const callback = this._getVOICallback(volumeId, displaySetOptions);
+
+      viewportVolumeInputs[volumeId] = {
+        imageIds,
+        volumeId,
+        callback,
+        blendMode: displaySetOptions.blendMode,
+        slabThickness: displaySetOptions.blendMode
+          ? displaySetOptions.slabThickness || 500
+          : undefined,
+      };
+    }
+
+    // Todo: this is only for the first load from hanging protocol, if
+    // viewport gets a new display set (from drag and drop) we don't want to
+    // use the hanging protocol image loading strategy
+    // const hasCustomLoad = this.HangingProtocolService.hasCustomImageLoadStrategy();
+
+    // If the hangingProtocol has specified a custom image loading strategy, use it.
+    // Otherwise, use the default image loading strategy which is based on the
+    // sorted imageIds in the volume. In addition, if displaysets has been
+    // invalidated and need to be re-rendered, we need to use the default image
+    // loading strategy.
+    // const displaySetShouldReRender = displaySets.some(displaySet => {
+    //   return this.displaySetsNeedRerendering.has(
+    //     displaySet.displaySetInstanceUID
+    //   );
+    // });
+
+    // if (!displaySetShouldReRender && hasCustomLoad) {
+    //   return;
+    // }
+
+    viewportData.volumes.forEach(volume => {
+      volume.load();
+    });
+
+    // this.initiateLoad(viewport.uid);
+
+    // const volumeInputs = this.sceneVolumeInputs.get(sceneUID);
```

Comment:
I can't tell how much of this commented code is WIP and required, or should be deleted if unused?